extensions/kadena/src/extension.ts

```diff
@@ -0,0 +1,147 @@
+import { ITatumSdkContainer, Network, TatumConfig, TatumSdkExtension } from '@tatumio/tatum'
+import {
+  ChainId,
+  createClient,
+  IUnsignedCommand,
+  Pact,
+  PactReference,
+  readKeyset
+} from "@kadena/client";
+
+import { KadenaGuard, KadenaNetworkId } from "./types";
+import {
+  COIN_GAS_CAPABILITY,
+  CREATE_TOKEN_ID_POLICIES,
+  NFT_PRECISION,
+  MARMALADE_CREATE_TOKEN_CAPABILITY,
+  MARMALADE_MINT_CAPABILITY,
+  NFT_AMOUNT,
+  MARMALADE_BURN_CAPABILITY,
+  NFT_BURN_AMOUNT,
+  MARMALADE_BURN_POLICY
+} from "./consts";
+import { KadenaLoadBalancerRpc } from "@tatumio/tatum/dist/src/service/rpc/other/KadenaLoadBalancerRpc";
+
+export class KadenaUtils extends TatumSdkExtension {
+  private readonly kadenaRpc: KadenaLoadBalancerRpc
+  private readonly sdkConfig: TatumConfig
+
+  constructor(tatumSdkContainer: ITatumSdkContainer) {
+    super(tatumSdkContainer)
+    this.sdkConfig = this.tatumSdkContainer.getConfig()
+    this.kadenaRpc = this.tatumSdkContainer.get(KadenaLoadBalancerRpc)
+  }
+
+  public async mintBasicNFT(uri: string | PactReference, publicKey: string, chainId: ChainId) : Promise<IUnsignedCommand> {
+
+    const guard : KadenaGuard = this.getGuardFromKey(publicKey)
+    const {networkId, client} = this.getTatumKadenaClient(chainId);
+
+    const createTokenIdCommand = Pact.builder
+      .execution(
+        Pact.modules['marmalade-v2.ledger']['create-token-id'](
+          {precision: NFT_PRECISION, uri, policies: () => CREATE_TOKEN_ID_POLICIES},
+          readKeyset('guardCreateTokenId')
+        )
+      )
+      .addKeyset('guardCreateTokenId', guard.keyset.pred, ...guard.keyset.keys)
+      .setMeta({
+        chainId: chainId,
+        senderAccount: guard.account,
+      })
+      .setNetworkId(networkId)
+      .createTransaction();
+
+    const tokenIdResult = await client.dirtyRead(createTokenIdCommand);
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const tokenId = tokenIdResult.result['data'];
+
+    if(!tokenId) {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      throw new Error(`Unable to create TokenId: ${tokenIdResult.result['error']['message']}`);
+    }
+
+    return Pact.builder
+      .execution(
+        Pact.modules['marmalade-v2.util-v1']['mint-basic-NFT'](
+          uri,
+          readKeyset('myUniqueKeysetForBasicNFTMint')
+        )
+      )
+      .addKeyset('myUniqueKeysetForBasicNFTMint', guard.keyset.pred, ...guard.keyset.keys)
+      .addSigner(guard.keyset.keys, (signFor) => [
+        signFor(COIN_GAS_CAPABILITY),
+        signFor(MARMALADE_CREATE_TOKEN_CAPABILITY, tokenId, guard.keyset),
+        signFor(MARMALADE_MINT_CAPABILITY, tokenId, guard.account, NFT_AMOUNT )
+      ])
+      .setMeta({
+        chainId: chainId,
+        senderAccount: guard.account,
+      })
+      .setNetworkId(networkId)
+      .createTransaction();
+  }
+
+  private getTatumKadenaClient(chainId: ChainId) {
```

Comment:
Why keep it `private`? This function on its own could also be very useful.

Reply:
it's mostly because till we have server side routing/load-balancing we need to instantiate client for every request to make use it's using up to date URL :/

Reply:
Sorry, I don't understand how that is relevant. I'm not telling you to remove or rewrite it, just to make it `public` so it can be used on its own by people who want a different `Pact` method, but with our Tatum RPC.